### PR TITLE
bpo-28953: Use `raise from` when raising new IncompleteRead

### PR DESCRIPTION
**Pull request by 180909**
https://bugs.python.org/issue28953

https://bugs.python.org/issue28953

Automerge-Triggered-By: GH:asvetlov

### CODE REVIEW

**Comment by cool-RR** [comment]
Thank you Jiahua!

**Review by cool-RR — APPROVED** [review]

**Comment by cool-RR** [comment]
@180909 Please edit the PR's title, it's bad. You could change it to "bpo-28953: Use `raise from` when raising new IncompleteRead" or anything else that makes sense.

**Comment by cool-RR** [comment]
Also, for the core developer who's going to review this: Please add the "skip news" label.

File: Lib/http/client.py [hunk 1]
```diff
@@ -593,8 +593,8 @@ def _read_chunked(self, amt=None):
                     amt -= chunk_left
                 self.chunk_left = 0
             return b''.join(value)
-        except IncompleteRead:
-            raise IncompleteRead(b''.join(value))
+        except IncompleteRead as incomplete_read:
+            raise IncompleteRead(b''.join(value)) from incomplete_read
```
Comment (asvetlov) [inline Lib/http/client.py]:
```suggestion
        except IncompleteRead as exc:
            raise IncompleteRead(b''.join(value)) from exc
```
follow (informal) code conventions used by stdlib

**Review by asvetlov — APPROVED** [review]

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**